python_modules/dagster/dagster_tests/asset_defs_tests/test_internal_freshness.py

```diff
@@ -1,170 +1,302 @@
-        assert isinstance(policy, TimeWindowFreshnessPolicy)
-        assert policy.fail_window == SerializableTimeDelta.from_timedelta(timedelta(minutes=10))
-        assert policy.warn_window == SerializableTimeDelta.from_timedelta(timedelta(minutes=5))
-
-
-def test_internal_freshness_policy_fail_window_validation() -> None:
-    with pytest.raises(CheckError):
-        InternalFreshnessPolicy.time_window(fail_window=timedelta(seconds=59))
-
-    with pytest.raises(CheckError):
+    def test_time_window_freshness_policy_fail_window_validation(self) -> None:
+        with pytest.raises(CheckError):
+            InternalFreshnessPolicy.time_window(fail_window=timedelta(seconds=59))
+
+        with pytest.raises(CheckError):
+            InternalFreshnessPolicy.time_window(
+                fail_window=timedelta(seconds=59), warn_window=timedelta(seconds=59)
+            )
+
+        # exactly 1 minute is ok
+        InternalFreshnessPolicy.time_window(fail_window=timedelta(seconds=60))
         InternalFreshnessPolicy.time_window(
-            fail_window=timedelta(seconds=59), warn_window=timedelta(seconds=59)
+            fail_window=timedelta(seconds=61), warn_window=timedelta(minutes=1)
         )
 
-    # exactly 1 minute is ok
-    InternalFreshnessPolicy.time_window(fail_window=timedelta(seconds=60))
-    InternalFreshnessPolicy.time_window(
-        fail_window=timedelta(seconds=61), warn_window=timedelta(minutes=1)
-    )
+
+class TestCronFreshnessPolicy:
+    def test_cron_freshness_policy_validation_basic(self) -> None:
+        """Can we define a cron freshness policy with valid parameters?"""
+        # Valid cron string and lower bound delta
+        policy = InternalFreshnessPolicy.cron(
+            deadline_cron="0 10 * * *",
+            lower_bound_delta=timedelta(hours=1),
+        )
+        assert isinstance(policy, CronFreshnessPolicy)
+        assert policy.deadline_cron == "0 10 * * *"
+        assert policy.lower_bound_delta == timedelta(hours=1)
+        assert policy.timezone == "UTC"
+
+    def test_cron_freshness_policy_validation_with_timezone(self) -> None:
+        policy = InternalFreshnessPolicy.cron(
+            deadline_cron="0 10 * * *",
+            lower_bound_delta=timedelta(hours=1),
+            timezone="America/New_York",
+        )
+        assert isinstance(policy, CronFreshnessPolicy)
+        assert policy.deadline_cron == "0 10 * * *"
+        assert policy.lower_bound_delta == timedelta(hours=1)
+        assert policy.timezone == "America/New_York"
+
+    def test_cron_freshness_policy_validation_invalid_cron(self) -> None:
+        with pytest.raises(CheckError):
+            InternalFreshnessPolicy.cron(
+                deadline_cron="0 10 * * * *",  # we don't support seconds resolution in the cron
+                lower_bound_delta=timedelta(hours=1),
+            )
+
+    def test_cron_freshness_policy_validation_invalid_timezone(self) -> None:
+        with pytest.raises(CheckError):
```

Comment:
nit - would be nice to assert on the error message thrown, since it's a custom message